corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -455,15 +514,19 @@ def get_cached_payload(self):
         else:
             payload = self.cache.get(self._initial_cache_key())
 
-        if payload and payload.endswith(FileRestoreResponse.EXTENSION) and not path.exists(payload):
-            return
-        return payload
+        return CachedResponse(payload)
 
     def set_cached_payload_if_necessary(self, resp, duration):
+        cache_payload = resp.get_cache_payload(bool(self.sync_log))
         if self.sync_log:
             # if there is a sync token, always cache
             try:
-                self.sync_log.set_cached_payload(resp, self.version)
+                data = cache_payload['data']
```

Comment:
What was the decision for setting the cache payload differently in the two types of stores? It looks like the sync log takes just the data whereas redis takes the object with the `is_file` and `data` attributes.